src/rgw/rgw_lua.cc

```diff
@@ -70,188 +65,53 @@ std::string script_oid(context ctx, const std::string& tenant) {
-        if (ret < 0) {
-            return ret;
-        }
-    }
-  }
-  return 0;
-}
-
-int list_packages(const DoutPrefixProvider *dpp, rgw::sal::RadosStore* store, optional_yield y, packages_t& packages) {
-  constexpr auto max_chunk = 1024U;
-  std::string start_after;
-  bool more = true;
-  int rval;
-  while (more) {
-    librados::ObjectReadOperation op;
-    packages_t packages_chunk;
-    op.omap_get_keys2(start_after, max_chunk, &packages_chunk, &more, &rval);
-    const auto ret = rgw_rados_operate(dpp, *(store->getRados()->get_lc_pool_ctx()),
-      PACKAGE_LIST_OBJECT_NAME, &op, nullptr, y);
-  
-    if (ret < 0) {
-      return ret;
-    }
+int remove_package(const DoutPrefixProvider *dpp, rgw::sal::Store* store, optional_yield y, const std::string& package_name)
+{
+  auto lua_mgr = store->get_lua_manager();
 
-    packages.merge(packages_chunk);
-  }
- 
-  return 0;
+  return lua_mgr->remove_package(dpp, y, package_name);
 }
 
-int install_packages(const DoutPrefixProvider *dpp, rgw::sal::RadosStore* store, optional_yield y, packages_t& failed_packages, std::string& output) {
-  // luarocks directory cleanup
-  std::error_code ec;
-  const auto& luarocks_path = store->get_luarocks_path();
-  if (std::filesystem::remove_all(luarocks_path, ec)
-      == static_cast<std::uintmax_t>(-1) &&
-      ec != std::errc::no_such_file_or_directory) {
-    output.append("failed to clear luarock directory: ");
-    output.append(ec.message());
-    output.append("\n");
-    return ec.value();
-  }
-
-  packages_t packages;
-  auto ret = list_packages(dpp, store, y, packages);
-  if (ret == -ENOENT) {
-    // allowlist is empty 
-    return 0;
-  }
-  if (ret < 0) {
-    return ret;
-  }
-  // verify that luarocks exists
-  const auto p = bp::search_path("luarocks");
-  if (p.empty()) {
-    return -ECHILD;
-  }
-
-  // the lua rocks install dir will be created by luarocks the first time it is called
-  for (const auto& package : packages) {
-    bp::ipstream is;
-    const auto cmd = p.string() + " install --lua-version " + CEPH_LUA_VERSION + " --tree " + luarocks_path + " --deps-mode one " + package;
-    bp::child c(cmd, bp::std_in.close(), (bp::std_err & bp::std_out) > is);
-
-    // once package reload is supported, code should yield when reading output
-    std::string line = std::string("CMD: ") + cmd;
+int list_packages(const DoutPrefixProvider *dpp, rgw::sal::Store* store, optional_yield y, packages_t& packages)
+{
+  auto lua_mgr = store->get_lua_manager();
 
-    do {
-      if (!line.empty()) {
-        output.append(line);
-        output.append("\n");
-      }
-    } while (c.running() && std::getline(is, line));
+  return lua_mgr->list_packages(dpp, y, packages);
+}
 
-    c.wait();
-    if (c.exit_code()) {
-      failed_packages.insert(package);
-    }
-  }
+int install_packages(const DoutPrefixProvider *dpp, rgw::sal::Store* store, optional_yield y, packages_t& failed_packages, std::string& output)
```

Comment:
this does not have any RADOS dependency.
it just calls into `list_packages()`

Reply:
I don't understand what you're trying to tell me here.